submissions/5mountains/dom-api-task/js/main.js

```diff
@@ -0,0 +1,104 @@
+const styles = {
+    menuLink: 'menu__link',
+    menuItem: 'menu__item',
+    heroItem: 'hero__item',
+    heroImg: 'hero__image',
+    itemIcon: 'item__icon',
+    itemTitle: 'item__caption',
+    itemPrice: 'item__price',
+    itemInfo: 'item__description'
+};
+
+async function fetchAsync () {
+    let response = await fetch('./data/data.json');
+    let data = await response.json();
+    return data;
+}   
+  
+fetchAsync()
+    .then(data => {
+        let details = createHtml(data);
+        drawPage(details);
+        toggleTabs();
+        animation();
+    })
+    .catch(reason => console.log(reason.message));
+
+function createHtml(data) {
+    const menuItems = document.createDocumentFragment();
+    const heroItems = document.createDocumentFragment();
+    data.pancakes.forEach((elem, index) => {
+        const element = elem;
+        const token = index;
+        const navContent = generateNavContent(element, token); 
+        const heroContent = generateHeroContent(element, token); 
+        menuItems.appendChild(navContent);
+        heroItems.appendChild(heroContent);
+    });
+    return {
+        menuItems,
+        heroItems
+    };
+}
+
+function generateNavContent(el, i) {
```

Comment:
Please, give a full name to the variable `i` :)

Such naming, with one letter, is antipattern.

Reply:
okay, good notice!